auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,197 @@
+                    "id": flag["id"],
+                    "requestee": flag["requestee"],
+                    "requestee_status": user_activity.get_string_status(
+                        inactive_users[flag["requestee"]]
+                    ),
+                }
+                for flag in bug["needinfo_flags"]
+                if flag["requestee"] in inactive_users
+            ]
+            bug["inactive_ni_count"] = len(bug["inactive_ni"])
+            self.add_action(bug)
+
+        return bugs
+
+    @staticmethod
+    def should_forward_needinfo(bug):
+        """
+        Determain if the bug is important enough to have the needinfos forwarded
+        to the triage owner.
+        """
+
+        return (
+            bug["priority"] in HIGH_PRIORITY
+            or bug["severity"] in HIGH_SEVERITY
+            or lmdutils.get_date_ymd(bug["creation_time"]) >= RECENT_BUG_LIMIT
+        )
+
+    def add_action(self, bug):
+        users_num = len(set([flag["requestee"] for flag in bug["inactive_ni"]]))
+
+        if self.should_forward_needinfo(bug):
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "status": "?",
+                        "requestee": bug["triage_owner"],
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Redirect needinfo pending on inactive {utils.plural('user', users_num)} to the triage owner.",
+                },
+            }
+        else:
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "status": "X",
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Clear needinfo pending on inactive {utils.plural('user', users_num)}.",
+                },
+            }
+
+        self.add_prioritized_action(bug, bug["triage_owner"], autofix=autofix)
+
+    def get_bug_sort_key(self, bug):
+        return (
+            not self.should_forward_needinfo(bug),
+            utils.get_sort_by_bug_importance_key(bug),
+        )
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        data[bugid] = {
+            "priority": bug["priority"],
+            "severity": bug["severity"],
+            "creation_time": bug["creation_time"],
+            "triage_owner": bug["triage_owner"],
+            "needinfo_flags": [
+                flag for flag in bug["flags"] if flag["name"] == "needinfo"
+            ],
+        }
+
+        return bug
+
+    def get_bz_params(self, date):
+        date = lmdutils.get_date_ymd(date)
+
+        fields = [
+            "triage_owner",
+            "flags",
+            "priority",
+            "severity",
+            "creation_time",
+        ]
+
+        params = {
```

Comment:
Looking at one of the results, https://bugzilla.mozilla.org/show_bug.cgi?id=1767993, it makes me think that we should exclude recent needinfos (unless the needinfo is to a disabled account).
It happens sometimes that somebody might ask for some information from users that are seldomly active (e.g. in this case somebody from another company that is only needinfoed when something is broken in the integration of the other company's software in Firefox). We want to allow them at least a couple of weeks to receive the email and respond.

Reply:
I agree. We could just comment to inform in case of a recent needinfo (without doing any action). After it passes a time threshold we do the proper action (forward or clear the ni).

@marco-c wdyt?

Reply:
Yes, this is correct. I filed an issue (#1436) to follow up on this (it will be nice to have). For now, it is mitigated in 552f9ae524507dd61d4e451f068a700087acc644 to exclude recent needinfos (3 weeks).

Do you think the 3 weeks threshold is enough?

Reply:
Yes, it sounds good to me. Maybe even 2 weeks would be enough?
Could you add a comment to explain the reasoning, so we don't forget it in the future and drop the check?

Reply:
> Yes, it sounds good to me. Maybe even 2 weeks would be enough?

Done in c26e9b2ca5917f4ef64d061ee8f1427933d3e96e

> Could you add a comment to explain the reasoning, so we don't forget it in the future and drop the check?

We already have a comment in place:
https://github.com/mozilla/relman-auto-nag/blob/7d833d5567674eefeca20f113a18871474b18bda/auto_nag/scripts/inactive_ni_pending.py#L107-L113